Chessington.GameEngine/Pieces/Pawn.cs

```diff
@@ -15,27 +15,45 @@ public IEnumerable<Square> PawnMove(Board board, int direction, int startingRow)
             var currentRow = currentPos.Row;
             var currentCol = currentPos.Col;
 
-            if (!Square.At(currentRow + direction, currentCol).IsOccupied(board))
+            if (Square.At(currentRow + direction, currentCol).CanMoveTo(board))
             {
                 yield return Square.At(currentRow + direction, currentCol);
 
                 if (currentRow == startingRow &&
-                    !Square.At(currentRow + 2 * direction, currentCol).IsOccupied(board))
+                    Square.At(currentRow + 2 * direction, currentCol).CanMoveTo(board))
```

Comment:
this startingRow variable seems unnecessary.

You can use MoveTo method in Piece class and set a flag hasMoved on a piece.
And then check in Pawn (which is a subclass of a Piece)  if not hasMoved then allow 2 moves